dist/mjs/helper/util.js

```diff
@@ -53,66 +53,64 @@ export class ZkWasmUtil {
         }
-        // Convert value byte array
+        // Check if value is a hexdecimal
+        if (!re.test(value.slice(2))) {
+            throw new Error("Invalid hex value: " + value);
+        }
+        // Check the length of the hexdecimal is even
+        if (value.length % 2 != 0) {
+            throw new Error("Odd Hex length provided: " + value);
+        }
+        return true;
+    }
+    static validateInput(input) {
+        let inputSplit = input.split(":");
+        // Check that there are two parts
+        if (inputSplit.length != 2) {
+            throw new Error("Invalid input: " + input);
+        }
+        let value = inputSplit[0];
+        let type = inputSplit[1];
+        let decimlaRegex = new RegExp(/^\d+$/); // decimal
         if (type == "i64") {
-            let v;
+            // If 0x is present, check that it is a hexdecimal
             if (value.slice(0, 2) == "0x") {
-                v = new BN(value.slice(2), 16);
+                this.validateHex(value);
             }
+            // If 0x is not present, check that it is a decimal
             else {
-                v = new BN(value);
+                if (!decimlaRegex.test(value)) {
+                    throw new Error("Invalid input value: " + input);
+                }
             }
-            return [v];
         }
         else if (type == "bytes" || type == "bytes-packed") {
-            if (value.slice(0, 2) != "0x") {
-                console.log("Error: bytes input need start with 0x");
-                return null;
-            }
-            let bytes = ZkWasmUtil.hexToBNs(value.slice(2));
-            return bytes;
+            this.validateHex(value);
         }
         else {
-            console.log("Unsupported input data type: %s", type);
-            return null;
+            throw new Error("Invalid input type: " + type);
         }
     }
-    /**
-     * TODO: \
-     * This function may have some problem parsing bytes-packed, need to be fixed.
-     * Do not rely on this function to parse bytes-packed and verify proof information
-     */
-    static parseArgs(raw) {
-        let parsedInputs = new Array();
-        for (var input of raw) {
-            input = input.trim();
-            if (input !== "") {
-                let args = ZkWasmUtil.parseArg(input);
-                if (args != null) {
-                    parsedInputs.push(args);
-                }
-                else {
-                    throw Error(`invalid args in ${input}`);
-                }
-            }
+    // Inputs are strings that should be of the form 32:i64, 0x1234:bytes or 0x1234:bytes-packed and split by spaces
+    static validateInputs(inputs) {
+        let trimmed = inputs.trim();
+        if (trimmed === "") {
+            return [];
         }
-        return parsedInputs.flat();
+        // Split the inputs by spaces
+        let inputArray = trimmed.split(" ");
```

Comment:
<img width="501" alt="image" src="https://github.com/DelphinusLab/zkWasm-service-helper/assets/22124307/690c7e34-f9f6-4720-af23-885db7ee733f">
Looks like it cannot handle muptiple space.